### DISCUSSION TITLE
CronExpression.IsValidExpression validation

### DISCUSSION THREAD

**Discussion by oslanecr** [quartznet/quartznet#1402] [Q&A] [answered]
Hi all,

I started using the CronExpression.IsValidExpression function and found that in a very specific case the function returns 'true' even though the cron expression is not valid. Please see below for examples:

1) CronExpression.IsValidExpression("0 59 14 * * ?") returns 'true'
2) CronExpression.IsValidExpression("0 60 14 * * ?") returns 'false'
3) CronExpression.IsValidExpression("0 98 14 * * ?") returns 'false'
4) CronExpression.IsValidExpression("0 99 14 * * ?") **returns 'true' <-- this is not expected**
5) CronExpression.IsValidExpression("0 100 14 * * ?") returns 'false'

Could you, please, help me understand why '99' is accepted as valid minutes? The GetExpressionSummary() method returns "Minutes: *" in that case. Not sure what I am doing wrong.

Thank you in advance,
Robert

**Comment by lahma** [comment, marked as answer]
Probably stems with the fact that internally Quartz uses `99` as `*` value, so something is leaking trough the validation logic.

https://github.com/quartznet/quartznet/blob/6186e9e5e2c747b537fc5172ddbd0cad57ea3222/src/Quartz/CronExpression.cs#L250-L253

**Reply by oslanecr** [reply]
Thank you @lahma  for clarification. We will make sure we catch that before passing the expression into Quartz. 
Would you know if there is plan to remediate that in the code?
Thank you,
Robert

**Reply by lahma** [reply]
I would be glad to accept a PR with a test case (and the changes) showing this has been fixed 😉
